CleanWeather/Helpers/SharedViewController.swift

```diff
@@ -8,25 +8,6 @@
 
 import UIKit
 
-class SharedViewController: UIViewController {
-    
-    func presentSpinner(_ state: Bool) {
-        if state {
-            let spinner = SpinnerController()
-            guard navigationController?.presentedViewController == nil else { return }
-            navigationController?.present(spinner, animated: false, completion: nil)
-        } else {
-            if let spinner = navigationController?.presentedViewController as? SpinnerController {
-                spinner.dismiss(animated: false, completion: nil)
-            }
-        }
-    }
-    
-    func presentAlert(title: String, message: String) {
-        let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
-        let action = UIAlertAction(title: R.string.localizable.ok(), style: .default, handler: nil)
-        alert.addAction(action)
-        navigationController?.present(alert, animated: true, completion: nil)
-    }
-    
+class SharedViewController: UIViewController, SpinnerPresentable, AlertPresentable {
```

Comment:
No i nie rozumiem po co potrzebujemy tą pustą klasę. Czy nie można w takim wypadku jej zlikwidować i zrobić aby CityListController zamiast dziedziczyć z SharedViewController po prostu odwoływał sie do SpinnerPresentable i AlertPresentable?

Reply:
mozemy, ale po to robilismy to wszystko aby nie bylo konieczne robienie tego w akzdym mozliwym kontrolerze,

musialbys dopisac to do citylistcontrolera, i peoplelistcontrolera, i innych przyszlych
i tak jak wczesniej pisalem, kiedys proponowalem zeby wszystkie interactory, presentery itp mialy jakis wspolny mianownik [jakiegos wspolnego parenta] jakby trzebabylo dodac logike wszedzie